frontend/apps/crates/components/src/module/_groups/cards/edit/actions.rs

```diff
@@ -0,0 +1,111 @@
+use crate::module::_groups::cards::edit::state::*;
+use shared::domain::jig::module::body::{Background, ThemeChoice};
+use shared::domain::jig::module::body::_groups::cards::{CardPair as RawCardPair, Mode};
+use unicode_segmentation::UnicodeSegmentation;
+use utils::prelude::*;
+
+pub fn limit_text(max_len: usize, text: String) -> String {
+    let len = text.graphemes(true).count();
+
+    if len > max_len {
+        let cutoff_grapheme_byte = text.grapheme_indices(true).nth(max_len).unwrap_ji().0;
+
+        text[..cutoff_grapheme_byte].to_string()
+    } else {
+        text
+    }
+}
+
+impl<RawData: RawDataExt, E: ExtraExt> CardsBase<RawData, E> {
```

Comment:
Moved this implementation higher up the hierarchy so that it's accessible by all callers.